### PR TITLE
Push to elastic

### PR DESCRIPTION
**Pull request by KudumulaGangaNandini**
lint is :0
coverage is :85

### CODE REVIEW

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    @AfterReturning(pointcut="execution(* com.techsophy.tsf.runtime.form.service.impl.FormAclServiceImpl.saveFormAcl(..))",returning = "formAclDto")
+   public void afterSaveFormAclController(FormAclDto formAclDto)
+    {
+        if(formAclDto!=null)
+        {
+            Map<String,Object> map = new HashMap<>();
+            map.put("id",formAclDto.getId());
+            map.put("indexName",TP_RUNTIME_FORM_DATA+formAclDto.getFormId());
+            map.put("aclId",formAclDto.getAclId());
+            String token= tokenUtils.getTokenFromContext();
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: please use the new awgment commons

Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: updte to tag version for awgment-commons

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            var client = webClientWrapper.createWebClient(token);
+             webClientWrapper.webclientRequest(client,gatewayApi+"/elastic/v1/acl",POST,map);
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: please define constants in code.

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        }
+    }
+    @AfterReturning(pointcut="execution(* com.techsophy.tsf.runtime.form.service.impl.FormAclServiceImpl.deleteFormAcl(..)) && args(indexName)")
+    void afterDeleteFormAclController(String indexName)
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: the name of variable is wrong, its not an index name that comes in. Please confirm if you have ran it.

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

**Comment by VaibhavTechsophy** [comment]
has this been tested?

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @AfterReturning(pointcut="execution(* com.techsophy.tsf.runtime.form.service.impl.FormAclServiceImpl.saveFormAcl(..))",returning = "formAclDto")
+   public void afterSaveFormAclController(FormAclDto formAclDto)
+    {
+        if(formAclDto!=null)
+        {
+            Map<String,Object> map = new HashMap<>();
+            map.put(ID,formAclDto.getId());
+            map.put(INDEX_NAME,TP_RUNTIME_FORM_DATA+formAclDto.getFormId());
+            map.put(ACL_ID,formAclDto.getAclId());
+            String token= new com.techsophy.tsf.commons.user.UserDetails(gatewayApi).getToken().get();
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: this shoudl be injected as a bean

Comment (VaibhavTechsophy) [reply src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: please discuss with me

File: src/test/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    WebClientWrapper webClientWrapper;
+    @Mock
+    TokenUtils tokenUtils;
+    @InjectMocks
+    FormDataPushToElasticIndexName formDataPushToElasticIndexName;
+
+    @Test
+      void afterSaveFormAclController()
+    {
+        FormAclDto formAclDto = new FormAclDto();
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticTest.java]: add test case to cater to if formacldto is null

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Upgrade to latest released version of awgment commons

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            var client = webClientWrapper.createWebClient(token);
+            webClientWrapper.webclientRequest(client,gatewayApi+ELASTIC+VERSION_V1+ACL,POST,map);
+        }
+    }
+    @AfterReturning(pointcut="execution(* com.techsophy.tsf.runtime.form.service.impl.FormAclServiceImpl.deleteFormAcl(..)) && args(formId)")
+    void afterDeleteFormAclController(String formId)
+    {
+        String token= new com.techsophy.tsf.commons.user.UserDetails(gatewayApi).getToken().get();
+        var client = webClientWrapper.createWebClient(token);
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataPushToElasticIndexName.java]: donot use webclient wrapper here instead add to elastic service class

**Comment by VaibhavTechsophy** [comment]
@KudumulaGangaNandini  update to latest version - https://mvnrepository.com/artifact/com.techsophy/awgment-commons

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/ElasticAclAspect.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    private final FormDataElasticServiceImpl formDataElasticService;
+    private final ObjectMapper objectMapper;
+
+
+    @AfterReturning(pointcut="execution(* com.techsophy.tsf.runtime.form.service.impl.FormAclServiceImpl.saveFormAcl(..))",returning = "formAclDto")
+   public void afterSaveFormAclController(FormAclDto formAclDto)
+    {
+            Map<String,Object> map = new HashMap<>();
+            map.put(ID,formAclDto.getId());
+            map.put(INDEX_NAME,TP_RUNTIME_FORM_DATA+formAclDto.getFormId());
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/ElasticAclAspect.java]: Please keep this logic at a central utility

File: src/main/java/com/techsophy/tsf/runtime/form/dto/ELasticAcl.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.techsophy.tsf.runtime.form.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/dto/ELasticAcl.java]: dont use allargsconstructor

File: src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataElasticServiceImpl.java [hunk 9]
```diff
@@ -22,19 +30,42 @@ public class FormDataElasticServiceImpl implements FormDataElasticService
...
-            webClientWrapper.webclientRequest(webClientWrapper.createWebClient(tokenUtils.getTokenFromContext()),
-                    gatewayApi+ELASTIC_VERSION1+SLASH+ TP_RUNTIME_FORM_DATA +formDataDefinition.getFormId() +PARAM_SOURCE+elasticSource,POST,
-                    formDataDefinition);
+        if(elasticEnable) {
+            try {
+                webClientWrapper.webclientRequest(webClientWrapper.createWebClient(tokenUtils.getTokenFromContext()),
+                        gatewayApi + ELASTIC_VERSION1 + SLASH + TP_RUNTIME_FORM_DATA + formDataDefinition.getFormId() + PARAM_SOURCE + elasticSource, POST,
+                        formDataDefinition);
+            } catch (Exception e) {
+                log.error(e.getMessage());
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormDataElasticServiceImpl.java:52]: need review with @Mohanakhil

File: src/test/java/com/techsophy/tsf/runtime/form/aspect/ElasticAclAspectTest.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        FormAclDto formAclDto = new FormAclDto();
+        formAclDto.setId("12");
+        formAclDto.setFormId("23131");
+        formAclDto.setAclId("34");
+        ELasticAcl eLasticAcl = new ELasticAcl();
+        eLasticAcl.setId("12");
+        eLasticAcl.setIndexName("tp_runtime_form_data_23131");
+        eLasticAcl.setAclId("34");
+        Mockito.when(objectMapper.convertValue(any(),eq(ELasticAcl.class))).thenReturn(eLasticAcl);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/aspect/ElasticAclAspectTest.java]: this should not be mocked but rather verified that the dto is properly constrcuted

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java [hunk 11]
```diff
@@ -20,14 +28,16 @@
...
+    private UserDetails userDetails;
     private final String gatewayURL="http://localhost:8080";
     private final boolean elasticSource=true;
 
     @Test
     void saveOrUpdateToElasticTest()
     {
         FormDataElasticServiceImpl mockFormDataElasticServiceImpl=
-                new FormDataElasticServiceImpl(mockWebClientWrapper,mockTokenUtils,gatewayURL,elasticSource);
+                new FormDataElasticServiceImpl(mockWebClientWrapper,mockTokenUtils,gatewayURL,elasticSource,true,userDetails);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java]: remove all args constructur

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java [hunk 12]
```diff
@@ -37,9 +47,37 @@ void saveOrUpdateToElasticTest()
...
         Assertions.assertThrows(RuntimeException.class,()->mockFormDataElasticServiceImpl.saveOrUpdateToElastic(formDataDefinition));
     }
+    @Test
+    void saveACL()
+    {
+        ELasticAcl eLasticAcl = new ELasticAcl();
+        eLasticAcl.setId("12");
+        eLasticAcl.setIndexName("tp_runtime_form_data_12");
+        eLasticAcl.setAclId("123");
+        FormDataElasticServiceImpl mockFormDataElasticServiceImpl=
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java]: why is this being instantiated and not ibnjected?

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java [hunk 13]
```diff
@@ -37,9 +47,37 @@ void saveOrUpdateToElasticTest()
...
+                new FormDataElasticServiceImpl(mockWebClientWrapper,mockTokenUtils,gatewayURL,elasticSource,true,userDetails);
+        WebClient webClient= WebClient.builder().build();
+        when(mockWebClientWrapper.createWebClient(any())).thenReturn(webClient);
+        when(mockWebClientWrapper.webclientRequest(any(),anyString(),any(),any())).thenReturn("abc");
+        when(userDetails.getToken()).thenReturn(Optional.of("abc"));
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java:70]: verify if token is not present

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]
Pls provide the lint and coverage count
Also I asked you to change the version for awgment-commons?

**Review by VaibhavTechsophy — APPROVED** [review]

**Comment by VaibhavTechsophy** [comment]
pls deploy
